src/CommunityToolkit.Maui/Essentials/AppTheme/AppThemeObjectOfT.shared.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// Represents an object that is aware of the operating system theme.
+/// </summary>
+public abstract class AppThemeObject<T>
+{
+	/// <summary>
+	/// The <see cref="object"/> that is used when the operating system uses light theme.
+	/// </summary>
+	public T? Light { get; set; }
+
+	/// <summary>
+	/// The <see cref="object"/> that is used when the operating system uses dark theme.
+	/// </summary>
+	public T? Dark { get; set; }
+
+	/// <summary>
+	/// The <see cref="object"/> that is used when the current theme is unspecified or
+	/// when a value is not provided for <see cref="Light"/> or <see cref="Dark"/>.
+	/// </summary>
+	public T? Default { get; set; }
+
+	/// <summary>
+	/// Gets a bindable object which holds the diffent values for each operating system theme. 
+	/// </summary>
+	/// <returns>A <see cref="AppThemeBinding"/> instance with the respective theme values.</returns>
+	public virtual BindingBase GetBinding()
+	{
+		var binding = new AppThemeBinding();
+
+		if (Light is not null)
```

Comment:
Do we need a check for null? Is null a valid value? What if null is a value value for T and I want to set it after the previous value?

Reply:
Already answered here
https://github.com/CommunityToolkit/Maui/pull/1264#discussion_r1241127495

Reply:
This may be an issue if null is a valid value. We cannot set it after some "not null" value